pkg/overlay/client_test.go

```diff
@@ -128,6 +128,41 @@ func TestLookup(t *testing.T) {
 	}
 
 }
+func TestBulkLookup(t *testing.T) {
+	cases := []struct {
```

Comment:
There's a single case here so no need for tables. Also rather than test internal behavior, test for expected external behavior.

Reply:
What do you mean by internal vs external behavior?

Reply:
The internal behavior is how the thing under the test achieves it's outcome. The external behavior is about what is expected out of the thing under test.

Testing internal behaviors is usually fragile and low-risk, meaning there's little to gain from testing it. Similarly, internal behavior can change and break tests, while all code is still correct. Let's say you introduce a caching layer into `client` then the bulk lookup may not be called at all.

External behavior is usually higher risk. In this case, there are few things that should hold for BulkLookup.

1. The behavior under missing ids, invalid ids.
2. Behavior on `BulkLookup([]dht.NodeID{})`
3. Behavior on randomized ordering of ids. That you do get things back in the right order or mapping.
2.1. Insert test-data in random order
2.2. add missing and invalid id's your initial order
2.3. Repeat n times:
2.4.1. shuffle the items
2.4.2. call GetAll with the first `rand.Intn(n)` elements
2.4.3. check whether the result is the right one

As for the table vs none, there's always many possibilities in extensions, however most of the time possible extensions end up unused.